### DISCUSSION TITLE
inoremap <expr> j pumvisible() ? "\<C-n>" : "j"  doesn't work.

### DISCUSSION THREAD

**Discussion by owlx56** [JetBrains/ideavim#898] [Q&A]
After I add the config `inoremap <expr> j pumvisible() ? "\<C-n>" : "j"` to `.ideavimrc`, when I press key `j` the character “j” will appear for about 0.5 seconds then disappear.

https://github.com/JetBrains/ideavim/assets/105162283/84bbf88c-1666-4b5c-b6c0-b1ddebc5a951

**Comment by owlx56** [comment]
More details about issue: https://stackoverflow.com/questions/78569851/how-can-i-use-j-k-to-select-next-previous-completion-suggestions-in-intellij-ide

**Comment by citizenmatt** [comment]
IdeaVim doesn't currently support the `pumvisible` Vim function, so this mapping won't work. Following the StackOverflow link, I'm not aware of a way to make IdeaVim move the completion selection up/down with `j`/`k`, sorry. I'm also not entirely sure how it could work without a similar mapping. Since you're in insert mode, `j` will naturally be inserted as though typed.

**Reply by owlx56** [reply]
Thanks for your reply, maybe I have to use `ctrl-j` `shift-j` to replace `j`...
